cmake/dependencies/CMakeLists.txt

```diff
@@ -213,3 +213,39 @@ if(BUILD_Cbc)
 endif()
 
 set(CMAKE_CXX_STANDARD 17)
+
+# ##############################################################################
+# GLPK
+# ##############################################################################
+if(BUILD_GLPK)
+  message(CHECK_START "Fetching GLPK")
+  list(APPEND CMAKE_MESSAGE_INDENT "  ")
+
+  find_program(MAKE_EXE NAMES gmake nmake make)
+
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/GLPK.CMakeLists.txt.in
+          ${CMAKE_CURRENT_BINARY_DIR}/GLPK/CMakeLists.txt @ONLY)
+
+  execute_process(
+          COMMAND ${CMAKE_COMMAND} -H. -Bproject_build -G "${CMAKE_GENERATOR}"
+          RESULT_VARIABLE result
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/GLPK)
+  if(result)
+    message(FATAL_ERROR "CMake step for GLPK failed: ${result}")
+  endif()
+
+  execute_process(
+          COMMAND ${CMAKE_COMMAND} --build project_build
+          RESULT_VARIABLE result
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/GLPK)
+  if(result)
+    if (result EQUAL 2)
+      message(FATAL_ERROR "Build step for GLPK failed: ${result}; the GLPK build process may require root (sudo) to use ${MAKE_EXE}")
+    else()
+      message(FATAL_ERROR "Build step for GLPK failed: ${result}")
+    endif()
+  endif()
+
```

Comment:
Why not using FetchContent like others ?

Reply:
I couldn't find an official VCS for glpk (only the GNU ftp), and it needed some specific build steps

Reply:
let's cmake-ize it, (like I did for coin-or BTW).
FYI the point is not because I think CMake is superior or I want to increase my technical debt, it's mostly because I want to have cross compilation in a near future and having everything using FetchContent aka deps as sub-build (or or-tools becoming a super build for its deps) is to be able to reuse the cross toolchain setup for free since GLPK become part of the CMake build...
e.g. linaro arm64 cross toolchain
src: https://github.com/google/or-tools/runs/4137862333?check_suite_focus=true
script: https://github.com/google/or-tools/blob/stable/tools/cross_compile.sh

BTW I can work on this part
note: this is also the case for bazel but here is by construction of this google build system you must assimilate everything to bazel build...
cf: https://github.com/google/or-tools/blob/stable/bazel/glpk.BUILD (need to bump it BTW)

Reply:
I did try to look for some official sources of pre-built binaries so I could just `FetchContent` it, but it seems a prebuilt binary repo would need someone to git it, cmake-rize it, cross-compile it and above all, maintain it. Is there perhaps a different way to cmaker-ize it? I only ask for my edification, and I do realize that wrapping a `make` build in a cmake script is a bit of a hack